core/src/main/java/org/opensearch/sql/expression/function/UserDefinedFunctionHelper.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.function;
+
+import java.util.Collections;
+import org.apache.calcite.schema.ImplementableFunction;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.InferTypes;
+import org.apache.calcite.sql.type.SqlOperandMetadata;
+import org.apache.calcite.sql.type.SqlReturnTypeInference;
+import org.apache.calcite.sql.validate.SqlUserDefinedFunction;
+
+public interface UserDefinedFunctionHelper {
```

Comment:
Add doc

Reply:
Why a helper an interface? Can be `UserDefinedFunctionFactory` or `UserDefinedFunctionTemplate?